ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/SeedService.scala

```diff
@@ -33,38 +34,51 @@ object SeedService {
     seeding match {
       case Seeding.Strong => StrongRandom
       case Seeding.Weak => WeakRandom
-      case Seeding.Static => new StaticRandom("static random seed service")
+      case Seeding.Static => staticRandom("static random seed service")
     }
 
-  // Pseudo random generator seeded with high entropy seed.
-  // May block while gathering entropy from the underlying operating system
-  // Thread safe.
-  object StrongRandom extends SeedService {
-    override val nextSeed: () => crypto.Hash =
-      crypto.Hash.secureRandom(
-        crypto.Hash.assertFromByteArray(
-          SecureRandom.getInstanceStrong.generateSeed(crypto.Hash.underlyingHashLength)
-        )
-      )
+  /** Pseudo random generator seeded with high entropy seed.
+    * May block while gathering entropy from the underlying operating system
+    * Thread safe.
+    */
+  // lazy to avoid gathering unnecessary entropy.
+  lazy val StrongRandom = {
+    val logger = LoggerFactory.getLogger(this.getClass)
+    var done = false
+
+    new Thread {
+      override def run(): Unit = {
+        Thread.sleep(5 * 1000)
+        if (!done)
+          logger.info(
+            s"""Trying to gather entropy from the underlying operating system to initialized the contract ID seeding, but the entropy pool seems empty.
+               |On testing environment consider using the "${Seeding.Weak.name}" mode, that produces insecure contract IDs but does not block on startup.""".stripMargin
+          )
+        while ({ Thread.sleep(10 * 1000); !done })
+          logger.info("Still Trying to gather entropy to initialized the contract ID seeding...  ")
+      }
+    }.start()
+
+    val seed = SecureRandom.getInstanceStrong.generateSeed(crypto.Hash.underlyingHashLength)
+
+    done = true
```

Comment:
This is probably fine, but maybe have a look if `java.util.Timer` can solve this without having to juggle threads and mutable state. I would also recommend to use some duration type (both `java.time.Duration` and `scala.concurrent.Duration` work) instead of using raw numbers.